cryptographic_estimators/MQEstimator/MQAlgorithms/mht.py

```diff
@@ -65,41 +65,39 @@ def __init__(self, problem: MQProblem, **kwargs):
         self._n_reduced = n
         self._m_reduced = m
 
-    def time_complexity(self):
+    def _compute_time_complexity(self, parameters: dict):
         """
-        Return the time complexity of mht algorithm
+        Return the time complexity of the algorithm for a given set of parameters
 
         EXAMPLES::
 
             sage: from cryptographic_estimators.MQEstimator.MQAlgorithms.mht import MHT
             sage: from cryptographic_estimators.MQEstimator.mq_problem import MQProblem
             sage: E = MHT(MQProblem(n=183, m=12, q=4), w=2.8)
             sage: E.time_complexity()
-            24.628922047916475
+            26.628922047916475
         """
         n, m, _ = self.problem.get_problem_parameters()
         w = self.linear_algebra_constant()
         if is_power_of_two(self.problem.order_of_the_field()):
             time = 0
         else:
             time = m
-        if self.complexity_type == ComplexityType.ESTIMATE.value:
-            time += log2(m * n ** w)
-        else:
-            time += 0
+
+        time += log2(m * n ** w)
         return time
 
-    def memory_complexity(self):
+    def _compute_memory_complexity(self, parameters: dict):
         """
-        Return the memory complexity of MHT algorithm
+        Return the memory complexity of the algorithm for a given set of parameters
 
         EXAMPLES::
 
             sage: from cryptographic_estimators.MQEstimator.MQAlgorithms.mht import MHT
             sage: from cryptographic_estimators.MQEstimator.mq_problem import MQProblem
             sage: E = MHT(MQProblem(n=183, m=12, q=4), w=2.8)
             sage: E.memory_complexity()
-            18.61636217728924
+            19.61636217728924
```

Comment:
the estimate check has nothing to do here just return it. for the other case we have the compute_tilde_o_memory function..